### ISSUE TITLE
Mandatory charset in Accept header?

### ISSUE THREAD

**Issue by nitinprakash96** [fosskers/servant-xml#7]
Hello,

The current implementation is

```haskell
instance Accept XML where
    contentType _ = "application" M.// "xml" M./: ("charset", "utf-8")
```

Trying to parse just `application/xml` yields `UnsupportedContentType application/xml`. A lot of the backend APIs don't return charset in their accept headers. One example is Azure blob storage services.

I wonder if it's okay to provide the charset as an option in this library itself. One possible way to mitigate this could be something like this?

```haskell
import Data.List.NonEmpty as NE

instance Accept XML where
    contentTypes _ = 
        "application" M.// "xml" M./: ("charset", "utf-8") NE.:|
        [ "application" M.// "xml" ]
```

**Comment by fosskers** [comment]
I must have based the original implementation off this example: https://hackage.haskell.org/package/servant-0.20.1/docs/Servant-API.html#t:Accept

I see that the default implementation of `contentType` (no S) just pulls the `NE.head` from the list, so that should act as the default when we're _writing_ XML. Thus we'd set our charset properly, which the net seems to recommend.

Then I assume on parsing, it'll fallback to the second if the first doesn't work.

**Comment by fosskers** [comment]
Can you test https://github.com/fosskers/servant-xml/pull/8 and see if it solves your issue? If it does, I'll make a release.

**Comment by nitinprakash96** [comment]
You're right that charset should be set by the backend systems since it's recommended but unfortunately a lot of them don't.

> Can you test https://github.com/fosskers/servant-xml/pull/8 and see if it solves your issue?

It does. Thanks for the quick respone :)

**Comment by fosskers** [comment]
Released as `1.0.3`.
